ios/FluentUI/Core/Theme/Tokens/FontInfo.swift

```diff
@@ -17,47 +17,75 @@ public struct FontInfo {
     /// - Parameter weight: The weight to use for the font. Defaults to `.regular`.
     ///
     /// - Returns: A struct containing the information needed to create a font object.
-    public init(name: String? = nil, size: CGFloat, weight: Font.Weight = .regular) {
+    public init(name: String? = nil,
+                size: CGFloat,
+                weight: Font.Weight = .regular) {
         self.name = name
         self.size = size
         self.weight = weight
     }
 
+    static var sizeTuples: [(size: CGFloat, textStyle: Font.TextStyle)] = [
+        (34.0, .largeTitle),
+        (28.0, .title),
+        (22.0, .title2),
+        (20.0, .title3),
+        // Note: `17.0: .headline` is removed to avoid needing duplicate size key values.
+        // But it's okay because Apple's scaling curve is identical between it and `.body`.
+        (17.0, .body),
+        (16.0, .callout),
+        (15.0, .subheadline),
+        (13.0, .footnote),
+        (12.0, .caption),
+        (11.0, .caption2)
+    ]
+
+    public var textStyle: Font.TextStyle {
+        // Defaults to smallest supported text style for mapping, before checking if we're bigger.
+        var textStyle = Font.TextStyle.caption2
+        for tuple in Self.sizeTuples {
```

Comment:
Should we really be going through a for loop every time someone wants to get the textStyle from a FontInfo? Even if it's not an especially long loop.

Reply:
I agree that it's a little unusual. Would you prefer an additional layer of cache? I'm not sure that there's any measurable CPU hit for this constrained loop, but I can double-check.

Reply:
Took a trace, and it looks like we spend ~1ms of time in that method, for a workload with 1.18 seconds of CPU time (just scrolling a table view up and down, constantly regenerating cells). 

I'm leaning towards living with that, but I'm open to looking for further optimization if it feels appropriate.

Reply:
That seems lightweight enough that we're probably okay as is. Thanks for checking!

Reply:
Upon further thinking, I agree with @huwilkes's first point. Gonna test a simple `Dictionary` lookup cache and report back on perf impact.

Reply:
Okay, confirmed that there's no real benefit (the dictionary lookups end up taking roughly the same amount of time as the tiny constrained loop).